src/lib.rs

```diff
@@ -276,11 +282,42 @@ mod contract {
             // TODO(#59) ETH transfer
             todo!();
         } else {
-            // TODO(#51) ERC20 transfer
-            todo!();
+            let mut engine = Engine::new(predecessor_address());
+            engine.receive_erc20_tokens();
         }
     }
 
+    /// Deploy ERC20 token mapped to a NEP141
+    #[no_mangle]
+    pub extern "C" fn deploy_erc20_token() {
+        // Id of the NEP141 token in Near
+        let args: DeployErc20TokenArgs =
+            DeployErc20TokenArgs::try_from_slice(&sdk::read_input()).sdk_expect("ERR_ARG_PARSE");
+
+        let mut engine = Engine::new(predecessor_address());
+
+        let erc20_contract = include_bytes!("../etc/eth-contracts/res/EvmErc20.bin");
+        let deploy_args = ethabi::encode(&[
+            ethabi::Token::String("TestToken".to_string()),
+            ethabi::Token::String("TT".to_string()),
+            ethabi::Token::Uint(ethabi::Uint::from(0)),
+            ethabi::Token::Address(current_address().into()),
```

Comment:
These values look suspicious

Reply:
I think this is here just for testing purposes as for now.

Reply:
This is "correct". Anyone can deploy an erc20 token and we populate metadata with default values. Will change name from `TestToken` to `Empty`, and symbol from `TT` to `EMPTY`.  Decimals are `0`, and last field is admin of the contract.

Reply:
We need to implement a way for trustlessly bridge metadata to the contract. For now, I'll add a trusted setup, where user with FullAccessKey will be able to set the metadata, so we can populate metadata properly for initial ERC20